File: 3637-trionic-array-i/3637-trionic-array-i-Solution1.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        p, q = 0, len(nums) - 1

        while(p < len(nums) - 3 and nums[p+1] > nums[p]):
            p += 1

        if(p == 0):
            return False

        while(q > 2 and nums[q-1] < nums[q]):
            q -= 1

        if(q == len(nums) - 1 or q <= p):
            return False

        for i in range(p, q):
            if nums[i] <= nums[i+1]:
                return False

        return True
        

File: 3637-trionic-array-i/3637-trionic-array-i-Solution1.cs
public class Solution {
    public bool IsTrionic(int[] nums) {
        if(nums.Length < 4) return false;

        var p = 0;
        var q = nums.Length - 1;

        while(p < nums.Length - 3 && nums[p+1] > nums[p]) p++;

        if(p == 0) return false;

        while(q > 2 && nums[q-1] < nums[q]) q--;

        if(q == nums.Length -1) return false;

        if(p >= q) return false;

        for(var i = p; i < q; i++ )
            if(nums[i+1] >= nums[i]) return false;

        return true;
    }
}


// [1, 3, 2, 3]
//. 0. 1. 2. 3